In [1]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
# import train test split
from sklearn.model_selection import train_test_split
# Evaluate the model using various metrics
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, roc_auc_score, confusion_matrix

In [3]:
df = sns.load_dataset('titanic')
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
df['sex'] = le.fit_transform(df['sex'])
# impute null values
df['age'] = df['age'].fillna(df['age'].mean())
X = df[['age', 'sex']]
y = df['survived']
# train test split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [7]:
# logistic regression
lr = LogisticRegression()
lr.fit(X_train, y_train)
y_pred = lr.predict(X_test)

# evaluating the model
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
f1 = f1_score(y_test, y_pred)
roc_auc = roc_auc_score(y_test, y_pred)
cm = confusion_matrix(y_test, y_pred)

print('Evaluation Metrics:')
print('-------------------')
print('Accuracy:', accuracy)
print('Precision:', precision)
print('Recall:', recall)
print('F1 Score:', f1)
print('ROC AUC Score:', roc_auc)
print('Confusion Matrix:', cm)

Evaluation Metrics:
-------------------
Accuracy: 0.7821229050279329
Precision: 0.7536231884057971
Recall: 0.7027027027027027
F1 Score: 0.7272727272727273
ROC AUC Score: 0.7703989703989704
Confusion Matrix: [[88 17]
 [22 52]]


In [8]:
# SVC
svc = SVC()
svc.fit(X_train, y_train)
y_pred = svc.predict(X_test)

# evaluating the model
accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
f1 = f1_score(y_test, y_pred)
roc_auc = roc_auc_score(y_test, y_pred)
cm = confusion_matrix(y_test, y_pred)

print('Evaluation Metrics:')
print('-------------------')
print('Accuracy:', accuracy)
print('Precision:', precision)
print('Recall:', recall)
print('F1 Score:', f1)
print('ROC AUC Score:', roc_auc)
print('Confusion Matrix:', cm)

Evaluation Metrics:
-------------------
Accuracy: 0.6089385474860335
Precision: 0.8333333333333334
Recall: 0.06756756756756757
F1 Score: 0.12500000000000003
ROC AUC Score: 0.529021879021879
Confusion Matrix: [[104   1]
 [ 69   5]]


Evaluation Metrics (of LR):
-------------------
Accuracy: 0.7821229050279329
Precision: 0.7536231884057971
Recall: 0.7027027027027027
F1 Score: 0.7272727272727273
ROC AUC Score: 0.7703989703989704


Evaluation Metrics (SVC):
-------------------
Accuracy: 0.6089385474860335
Precision: 0.8333333333333334
Recall: 0.06756756756756757
F1 Score: 0.12500000000000003
ROC AUC Score: 0.529021879021879